p2p-specs/p2p-interface.md

```diff
@@ -165,26 +167,40 @@ For any optional queueing, Bundlers SHOULD maintain maximum queue sizes to avoid
 
 #### Global topics
 
-There are two primary global topics used to propagate user operations (`UserOperationWithEntryPoint`)
-and aggregate user operation hashes (`NewPooledUserOperationsHashes`) to all nodes on the network.
+The primary global topics used to propagate user operations to all nodes on the network is `UserOperationWithEntryPoint`.
 
 ##### `user_ops_with_entry_point`
 
 The `user_ops_with_entry_point` topic is the concatenation of EntryPoint address and UserOperation message serialized using SSZ
 
 The following validations MUST pass before forwarding the `user_ops_with_entry_point` on the network
-- TBD
-- TBD
-- TBD
+- [IGNORE] `verified_at_block_hash` is too far in the past.
+- [REJECT] If any of the sanity checks specified in the [EIP](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4337.md#client-behavior-upon-receiving-a-useroperation) fails.
+- [REJECT] If the simulated user operation fails.
+
+### Encodings
+
+Topics are post-fixed with an encoding. Encodings define how the payload of a gossipsub message is encoded.
+
+ssz_snappy - All objects are SSZ-encoded and then compressed with Snappy block compression. Example: The user_ops_with_entry_point topic string of the canonical mempool is /account_abstraction/<mempool_id>/user_ops_with_entry_point/ssz_snappy, the <mempool_id> is <> and the data field of a gossipsub message is an UserOpsWithEntryPoiint that has been SSZ-encoded and then compressed with Snappy.
+Snappy has two formats: "block" and "frames" (streaming). Gossip messages remain relatively small (100s of bytes to 100s of kilobytes) so basic snappy block compression is used to avoid the additional overhead associated with snappy frames.
+
+Implementations MUST use a single encoding for gossip. Changing an encoding will require coordination between participating implementations.
+
+### Mempool ID
 
-##### `pooled_user_ops_hashes`
+The metadata associated to each mempool that a bundler supports is documented and stored in IPFS (a copy of this is also suggested to be submitted to `eth-infinitism` Github repo). This IPFS hash of the file is called `mempool-id` and this is used as the topic for subscription in the bundlers. The proposed structure of the mempool metadata is as follows
 
-The `pooled_user_ops_hashes` topic is used solely for propagating to all the connected nodes on the networks. One or more UserOps that have appeared in the network and which have not yet been included in a block are propagated to a fraction of the nodes connected to the network.
+```json
+{
+  "chainId": 1,
+  "entryPointContract": "0x0", //TBD with EntryPointConctractAddress
+  "description": "This is the default/canonical mempool, which will be used by most bundlers on Ethereum Mainnnet",
+  "minimumStake": 0.0,
+}
+```
+The `mempool-id` of the canonical mempool is `TBD`.
```

Comment:
We defined mempool-id to be the IPFS hash of a description.  Won't the canonical mempool id just be a hash of the above json (for chainId 1), and similarly the canonical mempool id of other chains will be hashes of the same json but with different chainId?

Reply:
My intention was to replace `TBD` with the actual IPFS hash corresponding to the canonical mempool of Ethereum mainnet. @yoavw